### PR TITLE
add training code for reward model

### PR DESCRIPTION
**Pull request by theblackcat102**
trainer code to train a single score reward model. Currently support webgpt and raw datasets from humanfeed back summary by openai. See readme and rank_datasets.py for more details.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-  "python.formatting.provider": "black",
```
Comment (andreaskoepf) [inline .vscode/settings.json:2]: Sorry, but we require all contributors to use the same pre-commit rules.

**Review by andreaskoepf — CHANGES_REQUESTED** [review]
Thanks for the PR! First step: Please run/install `pre-commit`, it is mandatory for all code that enters this repo.

File: .vscode/settings.json (hunk 1 above)
Comment (theblackcat102) [reply .vscode/settings.json:2]: just updated, please revise

File: model/reward/instructor/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,140 @@
+aiohttp==3.8.3
+aiosignal==1.3.1
```
Comment (andreaskoepf) [inline model/reward/instructor/requirements.txt]: This is a pretty extensive requirements.txt. Are all these libraries really required for the training of the model?

File: model/reward/instructor/tests/test_dataset.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+from experimental_dataset import DataCollatorForSummaryScore, HFSummaryQuality
+from rank_datasets import DataCollatorForPairRank, HFSummary, WebGPT
+from torch.utils.data import DataLoader
+from transformers import AutoTokenizer
+
```
Comment (andreaskoepf) [inline model/reward/instructor/tests/test_dataset.py:6]: very useful file, a short docstring at the beginning would be nice to explain how it is used during dev/purpose (e.g.  dataloader test, batch-shape inspection)

**Review by andreaskoepf — APPROVED** [review]
Overall nice training code! Thanks a lot ... also for instantly responding to change requests on discord.

File: model/reward/instructor/requirements.txt (hunk 2 above)
Comment (andreaskoepf) [reply model/reward/instructor/requirements.txt]: was already fixed by @theblackcat102

File: .vscode/settings.json (hunk 1 above)
Comment (yk) [reply .vscode/settings.json:2]: I still see the provider as autopep8, when it should be `black`. do you maybe have a local commit that you didn't push yet?

**Review by yk — APPROVED** [review]
just reset the formatting provider in settings.json to black, otherwise LGTM, thank you very much!

**Comment by theblackcat102** [comment]
@yk  yeah, it's my problem. just reset the format setting
